### PR TITLE
Modify quickstart to work with Docker for Mac

### PR DESCRIPTION
**Pull request by xvrl**
Moved all networking to a user-defined network which comes with DNS resolution by default

If we confirm this also works on Docker for Windows, we can probably remove the remaining mentions of docker-machine as well.

### CODE REVIEW

File: docs/quickstart.rst [hunk 1]
```diff
@@ -29,7 +29,7 @@ Getting Started with Docker Compose
 
 Docker Compose is a powerful tool that enables you to launch multiple Docker images in a coordinated fashion.  It is ideal for platforms like Confluent.  Before you get started, you will need to install both the core `Docker Engine <https://docs.docker.com/engine/installation/>`_ and `Docker Compose <https://docs.docker.com/compose/install/>`_.  Once you've done that, you can follow the steps below to start up the Confluent Platform services.
 
-#. Create and configure the Docker Machine.
+#. Create and configure the Docker Machine (Windows Only)
```
Comment (joel-hamill) [inline docs/quickstart.rst]: i'd add this at the beginning: `**Windows Only:** Create and configure the Docker Machine.`

File: docs/quickstart.rst [hunk 2]
```diff
@@ -203,23 +203,17 @@ Getting Started with Docker Client
...
+++++++++++++++
 
-  .. sourcecode:: console
-
-    $ docker-machine create --driver virtualbox --virtualbox-memory 6000 confluent
-
-  .. note::
-
-    If you want to reuse a Docker Machine you've already created under the name "confluent" run: ``docker-machine start confluent`` instead.
+First, create the network to run the Confluent containers in.
```
Comment (joel-hamill) [inline docs/quickstart.rst]: suggest: `Create the Docker network that is used to run the Confluent containers in.`

File: docs/quickstart.rst [hunk 3]
```diff
@@ -203,23 +203,17 @@ Getting Started with Docker Client
...
 
-Next, configure your terminal window to attach it to your new Docker Machine:
+.. note::
+  This is required to enable DNS resolution across all our containers, since the default Docker network does not have DNS enabled.
```
Comment (joel-hamill) [inline docs/quickstart.rst]: suggest: 
```
.. important:: A Docker network is required to enable DNS resolution across your containers. The default Docker network does not have DNS enabled.
```

File: docs/quickstart.rst [hunk 4]
```diff
@@ -229,12 +223,12 @@ Start ZooKeeper. You'll need to keep this service running throughout, so use a d
...
     $ docker run -d \
-        --net=host \
+        --net=confluent \
         --name=zookeeper \
-        -e ZOOKEEPER_CLIENT_PORT=32181 \
+        -e ZOOKEEPER_CLIENT_PORT=2181 \
         confluentinc/cp-zookeeper:4.1.0
 
-  This command instructs Docker to launch an instance of the ``confluentinc/cp-zookeeper:4.1.0`` container and name it ``zookeeper``.  We also specify that we want to use host networking and pass in the required parameter for running Zookeeper: ``ZOOKEEPER_CLIENT_PORT``.  For a full list of the available configuration options and more details on passing environment variables into Docker containers, see the `configuration reference docs <configuration.html>`_.
+  This command instructs Docker to launch an instance of the ``confluentinc/cp-zookeeper:4.1.0`` container and name it ``zookeeper``.  We also specify that we want to use the network we just created, and pass in the required parameter for running Zookeeper: ``ZOOKEEPER_CLIENT_PORT``.  For a full list of the available configuration options and more details on passing environment variables into Docker containers, see the `configuration reference docs <configuration.html>`_.
```
Comment (joel-hamill) [inline docs/quickstart.rst]: suggest: 
```
Also, the Docker network ``confluent`` and the required ZooKeeper parameter ``ZOOKEEPER_CLIENT_PORT`` are specified.
```

File: docs/quickstart.rst [hunk 5]
```diff
@@ -524,7 +510,9 @@ This portion of the quick start provides an overview of how to use Confluent Con
...
     [2016-08-26 18:47:26,811] INFO Started @5211ms (org.eclipse.jetty.server.Server)
 
-  To see the Control Center UI, open the link http://<ip-of-docker-host>:9021 in your browser.  The Docker Host IP is displayed with the command ``docker-machine ip confluent``.  If your Docker daemon is running on a remote machine (such as an AWS EC2 instance), you'll need to allow TCP access to that instance on port 9021. This is done in AWS by adding a "Custom TCP Rule" to the instance's security group; the rule should all access to port 9021 from any source IP.
+  To see the Control Center UI, open the link http://localhost:9021 in your browser.
+  
+  If you are running Docker Machine, the UI would be running at http://<docker-host-ip>:9021 where the Docker Host IP is the address displayed by running the command ``docker-machine ip confluent``.  If your Docker daemon is running on a remote machine (such as an AWS EC2 instance), you'll need to allow TCP access to that instance on port 9021. This is done in AWS by adding a "Custom TCP Rule" to the instance's security group; the rule should all access to port 9021 from any source IP.
```
Comment (joel-hamill) [inline docs/quickstart.rst]: suggest: `If you are running Docker Machine, the UI will be running at...`

File: docs/quickstart.rst [hunk 6]
```diff
@@ -901,4 +877,6 @@ Next you'll see how to monitor the Kafka Connect connectors in Control Center.
 Cleanup
 +++++++
 
-After you're done, cleanup is simple.  Run the command ``docker rm -f $(docker ps -a -q)`` to delete all the containers you created in the steps above for your target Docker Host.  Because you allowed Kafka and ZooKeeper to store data on their respective containers, there are no additional volumes to clean up.  If you also want to remove the Docker machine you used, you can do so using ``docker-machine rm <your machine name>``.
+After you're done, cleanup is simple.  Run the command ``docker rm -f $(docker ps -a -q)`` to delete all the containers you created in the steps above, ``docker volume prune`` to remove any remaining unused volumes, and ``docker network rm confluent`` to delete the network we created.
+
+If you are running Docker Machine, you may also want to remove the virtual machine you used. You can do so using ``docker-machine rm confluent``.
```
Comment (joel-hamill) [inline docs/quickstart.rst]: suggest: 
```
If you are running Docker Machine, you can remove the virtual machine with this command: ``docker-machine rm confluent``.
```

**Comment by xvrl** [comment]
@joel-hamill should be all set

**Comment by joel-hamill** [comment]
lgtm

**Pull request merged**